Sources/Sentry/SentryScope.m

```diff
@@ -412,6 +429,9 @@ - (BOOL)isEqualToScope:(SentryScope *)scope
         return NO;
     if (self.maxBreadcrumbs != scope.maxBreadcrumbs)
         return NO;
+    if (self.attachmentArray != scope.attachmentArray
```

Comment:
Needs exclusive access too.
Any access to `attachmentArray` now needs to take the lock

Reply:
I don't think it makes sense to add synchronization in here because it is not going to solve the initial problem:
```objc
if ([scopeA isEqualToScope:scopeB]) {
  // scopeA or scopeB is modified and they are not equal anymore
}
```
I think you can only solve the synchronization issue by getting a scope that for sure is not modified with `initWithScope`.
You should also use `initWithScope` if you would like to use the scope in combination with a hashtable. If not it could happen that you put the Scope into a hash table, keep the reference, and add for example a breadcrumb. Then you would get weird results because now the Scope in the hash table returns a different hash than it did when you added it. I opened a PR with comments to clarify that: https://github.com/getsentry/sentry-cocoa/pull/876 

I made a mistake in https://github.com/getsentry/sentry-cocoa/pull/743/ with adding the `isEqual` and `hashCode` implementation to the Scope. I would not like to add synchronized in here and remove these implementations with the next major. You were right back then.

Reply:
I also added a comment here (https://github.com/getsentry/sentry-cocoa/discussions/877#discussioncomment-207802) so we tackle it in v7.0.0

Reply:
you're reading the array outside of the lock and if its mutatated by another crash the program will crash